### ISSUE TITLE
cm01: Soliciting evaluation of discussions

### ISSUE THREAD

**Issue by vincenzocoia** [vincenzocoia/BAIT509#1]
Please put your comments here to the in-class exercise.

**Comment by chch1022** [comment]
https://stackoverflow.com/questions/48995299/im-trying-to-solve-a-puzzle-withcss

This is an ineffective question because:
1. the headline does not summarize the question 
2. in the text, he/she post the code they used but did not specify the question. No explanation of the code is given.

A better way to ask this question would be:
1. Explain the puzzle in details
2. Explain the code and give a context for solving the question

**Comment by Karansheraz** [comment]
Evaluating discussions:
link: https://github.com/scipy/scipy/issues/8480
The issue is worded effectively by clearly stating the issue in the title and providing minimal working code along with the question. Also, the author commented back when he successfully resolved the issue because of which he received an additional recommendation to tweak the code locally to get desired the result.

**Comment by kumar-srikanth** [comment]
Explain the situation, what you wanted to achieve, what is the code you are running and issue faced.

**Comment by iVibudh** [comment]
Evaluation Link : https://github.com/tidyverse/ggplot2/issues/2426

This is a simulation based on a real dataset, the comment suggests to use a self-contained reprex (short for minimal reproducible example). The comment also gives the plots to prevent any graphics related issue.

**Comment by karinatkach** [comment]
https://stackoverflow.com/questions/48989164/clustering-multivariate-time-series

This question is ineffective as there is no example of the code that would work and the issue is described very vaguely. To improve instead of using "for example, N patients" etc. a person could have used more precise data and provided a working example of the code that could be used to help.

**Comment by xinyizhoukoala** [comment]
https://stackoverflow.com/questions/12643009/regular-expression-for-floating-point-numbers
- The person who is asking the question makes his/her goal very clear, allowing others to know how to help him/her.
- This is an effective issue because the person who is asking the question shows the actual codes he/she writes in the programming tool, and shows the actual error given by the tool. 
- A better way could be to explain how he/she thinks through solving this particular problem, so people with more experience could offer more useful resources.

**Comment by nelsonwg** [comment]
https://github.com/tidyverse/ggplot2/issues/2464

The issue is worded effectively after the original poster pasted his question into the thread. The question provides some background to the problem and a minimum working example. The original poster then poses some objectives they would like to solve. 

The discussion has been supportive thus far. Comments provide potential solutions to the problems at hand and ask clarifying questions to better support the original poster.

**Comment by riowxm** [comment]
https://github.com/tidyverse/ggplot2/issues/2456
This is an effective issue because DesiQuintans states clearly that he/she wants to set the default aesthetics for a geom_histogram() and provided the code. He/she also provided the error message.
The good things here is that this person showed his/her guesses and efforts to try. Finally, reemphasize what exactly is confusing by raising specific questions.

**Comment by nibeerdewan** [comment]
Link to gghplot2 issue - https://github.com/tidyverse/ggplot2/issues/2429

About the question:
- The question is worded pretty concisely, in my opinion. But maybe a little too concise. I would suggest something like - "ggplot2: Color order for character variables in function scale_fill_manual()." This will enable people to look for keywords like ggplot2 and scale_fill_manual().
- Kudos for adding a minimum viable example. 

About the comments:
- Unfortunately, there is only one comment on this thread. This could be due to the lack of some keywords in the question or could be because this is a unique problem this user is facing?
- I like the response by MajoroMask. He suggests a solution and gives the 'why' for it.

**Comment by howardwu8** [comment]
https://github.com/numpy/numpy/issues/10601

The question is communicating with developers effectively. It is about how the Python  Numpy package gives vague error message in its histogram function when there is NaN element in the data. But the error message did not indicate so. 

The sample, though not a MWE, is good enough to describe the question and the issue. Since the error messages given from Python numpy package is not clear, the replying developer jumped in and submitted a maintenance change shortly. 

What could have been done better is to show MWE, or show the piece of code that the author had dug into the issues as he described. It is always better to show the code so that developers can save time on small maintenance.

**Comment by kishizhou** [comment]
- Link: https://github.com/scipy/scipy/issues/8465
- The issue clearly illustrates that the code that causes the problem. 
  He also reported his code and offered comment of his thinking. 
  At the same time, he provided the result of the code, error message and different version information. 
- The people who response is very supportive. He double confirmed the issues and clearly showed his code. More importantly, he added a reference at the last, it helps people who has further questions have more resources to look.

**Comment by FelixTrepanier** [comment]
scipy/scipy#8419

This is not really a question, but more of a discussion surrounding a bug of the Python spicy package. This initial post is very detailed, but is most likely too long. Indeed, it could most certainly be shorter, which would help people reproduce the bug.

Comments were constructive and attempted to bring value to the discussion. The first comment made by andyfaff is a good example of such comment.

**Comment by ChangHe591** [comment]
Issue posted online:
https://github.com/tidyverse/ggplot2/issues/2432

The question was not asked effectively since the code that could reproduce the issue was not included. There was one comment suggesting that the person should go ahead with set_last_plot() instead.
For the purpose of sharing/learning, it would be better for the owner to post if the new code helps or not.

**Comment by menggeyu** [comment]
Link: https://github.com/scipy/scipy/issues/8465

This issue is an effective question. It gives one example code of the problem and clarifies the error message and any Scipy/Numpy/Python version information.

In the comment, the discussion between not552 and jaimefrio is also effective. They both state their solutions clearly and include some possible code solution.

**Comment by fish117** [comment]
link:https://github.com/tidyverse/ggplot2/issues/2429

- This question is very clear and efficient because it describes the goal is assigning the color into the break. In addition, it added the code the pointed out his puzzle is how to transfer the variable into a factor or sth like that.

- The comments are very helpful because it pointed out by default ggplot 2 will use factor to solve this problem and showed the code of transferring the vector into factor.

- The improvement is he could show the warning information which could clarify the type of error and help people figure out this problem.

**Comment by Maggielena** [comment]
https://stackoverflow.com/questions/48995463/jenkins-nested-reactive-parameters-not-updating

This is an effective issue as the person clearly stated the problem and also included a valid minimum working example of his/her codes. 
- The layout of the issue is also clear
- The title explicitly named the issue

**Comment by jacobliu21** [comment]
Question posed about how to insert a list into numpy-based matrix: https://stackoverflow.com/questions/48913031/insert-a-list-into-numpy-based-matrix
The person who was asking the question made it clear what the goal was, how he or she made attempts, and what problems the person encountered. 
It might be better if he put the wrong outputs he got on the board as well to let people explain better from the wrong answer.
The only answer posted seems to be correct and very informative. It had a discussion on the outputs of different scenarios and actually provided more info than what had been asked.

**Comment by tinali92** [comment]
Question link: https://stackoverflow.com/questions/2620343/what-is-machine-learning

It is an effective question as the main issues are clearly stated in the title in an order of broader to narrower questions. People have a clear structure of how to answer the questions.

Some discussions was going on about the top comment. Kushal Waikar asked more information about the topic, and the author gave another paragraph to explain the topic, and the discussion added more avlue to the answer. 

The answers are almost all in long paragraph. I recoomend to use bullet points instead.

**Comment by luoxinru2017** [comment]
[https://github.com/tidyverse/ggplot2/issues/2363](url) 
This issue gives much value by clearly presenting the problem shares several ways in graphics device opening in both Rstudio and R.
The wording is concise and effective.

**Comment by satsuma24** [comment]
https://stackoverflow.com/questions/48766561/basemap-maskoceans-float-object-cannot-be-interpreted-as-an-integer

This is an effective issue based on key points below:
- The error is clearly stated in brief words on the first line.
- Version of Python is important here and it is stated.
- Python codes are shown with the type error for viewers to follow quickly.
- Full stacktrack is shown as well which means first-step error track is saved for answerers.

Improvements can be:
- Full error trace is necessary according to the discussion.
- Numbers of lines can be shown aside.

**Comment by Jimyie** [comment]
https://github.com/tidyverse/ggplot2/issues/2439

The question was worded effectively because it proposed the issue concisely in the title and include the graph on the main body of the explanation of the issue happened on his computer. 
Also, in this issue he also included the full codes which can let others see the question.

And someone proposed a idea that might solve this issue.

At last, he figured it out and posted the solution in this issue.

**Comment by sahilajmera** [comment]
https://github.com/tidyverse/ggplot2/issues/2452

The heading describes the issue well. The explanation provides a MWE which easily helps reproduce the error. The package version, software version and operating system are mentioned as well.
After a few discussions the issue was closed by the owner of the package, but the contributor added a reference to a similar issue and opened the issue again.

**Comment by arielshr** [comment]
https://stackoverflow.com/questions/48793896/writing-results-from-sql-query-to-csv-and-avoiding-extra-line-breaks

The person who raises the question clearly stated his/her problem with sample codes and thorough background and work descriptions.
The headline summarizes his/her problems appropriately.

Some of the comments are useful since they gave their own opinions and suggestions, even some helpful codes regarding the problem. However, there also exist some comments that doubt his concerns without really understanding his question.

**Comment by goyaldipin1989** [comment]
https://github.com/scipy/scipy/issues/8465

Effective question
The question though does not clearly explain about the unexpected behavior, However, in the detailed section, the writer explains about the asymmetric behavior and what was expected

The Minimum working code is shared, and some explanation of where the problem could be!

The comments offer some recommendation to try and solve the problem!

Improvements
If the unexpected behavior can be added to the question section!

**Comment by vnirwan** [comment]
Link: https://stackoverflow.com/questions/14604435/turning-off-some-legends-in-a-ggplot
This problem is well formulated and explains the question in a detailed manner. It also provided minimal writing code to help others to find out the answer.

**Comment by ishakataria** [comment]
Issue link: https://github.com/amueller/word_cloud/issues/70

Found it ineffective, the person only described his problem, without an example. Then one of the respondents who was referred to by the issue writer, requested for a minimal working example.

Despite the request, the author of the issue did not provide the minimal working code, and tried to explain it in his way, which might not have been comfortable for the respondent to follow. 

Consequently the discussion went very long, and the respondent could not understand the problem, and the author found his mistake himself!

**Comment by Pallavi-Agarwal** [comment]
Link: https://github.com/tidyverse/ggplot2/issues/2110
This is short but specific, has the code and keywords. And the solution is WIP

**Comment by souyazdi** [comment]
statsmodels/statsmodels

The question title and the explanation on the issue is clear. He included the code snippet that he believes that causes the error and he states what he wants to get  out of his attempt. Responses to the problem are relative and informative as people not only provided discussion on the problem based on their own knowledge but also they have provided useful links that they think might help with resolving the problem. Overall, I found this discussion effective.

**Comment by sliqiong** [comment]
Issue related to gglpot2: https://github.com/tidyverse/ggplot2/issues/2361

The question is worded effectively as it describes the situation that the author of the issue failed to add alpha to a gradient legend scale with the minimum working code. 

In response, one comment corrected the wrong judgement of the previous comment by clarifying one possible suggestion that alpha values can be set directly in the colors so that people will have more flexibility in many contexts than the ggplot2 alpha option provides.

In conclusion, the discussion is supportive and friendly.

**Comment by piyush7007** [comment]
https://stackoverflow.com/questions/10059594/a-simple-explanation-of-naive-bayes-classification

Go through the post to get a brief on naive bayes algorithm.

**Comment by harshclover** [comment]
https://www.kaggle.com/mrisdal/exploring-survival-on-the-titanic/code

Some of the best examples of Machine Learning. The Titanic Shipwreck Case! Give it a read.

**Comment by piyush7007** [comment]
Test Issue:

Evaluation Link : tidyverse/ggplot2#2426

This is a simulation based on a real dataset, the comment suggests to use a self-contained reprex (short for minimal reproducible example). The comment also gives the plots to prevent any graphics related issue.

**Comment by Coding-Cody** [comment]
https://github.com/scipy/scipy/issues/8419

The question is well structured with subtitles:
**Background
Data
A workaround
Potential fixes, work required**

The issuer state the objective and what he is doing clearly. A taste of the data is also offered to provide first-hand intuition. The question can add some hashtag to make it clear what keywords the question have before people click in the question page.

**Comment by dhruvatej** [comment]
Hey MBAN! Ever wondered how Facebook uses machine learning to automatically recognize your friends to tag in your Photographs? Give it a read to know more!

I have come across an interesting Issue: Improving detection of faces? #333 while skimming through the issues this morning. I couldn’t resist in deep diving into the concept to give you an idea of how it works.

In the old days, we used to type in the name to tag a friend. Now as soon as you upload a picture, it tags/prompts their names to tag for you. Face recognition is the technology used by Facebook, an interesting concept for detecting faces using Python.

Detection Accuracy Percentage: 98%

Steps involved in Face Recognition:

Find face in the image (Uses Algorithm-Histogram of Oriented Gradients(HOG): Check under references to learn more)
Analyze facial features and focus on each face (Albeit in Bad lighting) and recognize if it is the same person. Pick the unique features like how big the eyes are, how long the face is. (Finding the landmarks)
Compare against known faces. (Uses Neural Network to measure facial features)
And it’s a Match (Finds the name of a person using Classification and Simple Linear Regression Analysis by SVM Classifier Algorithm)
References
HOG - http://lear.inrialpes.fr/people/triggs/pubs/Dalal-cvpr05.pdf
HOG Python Code - https://gist.github.com/ageitgey/1c1cb1c60ace321868f7410d48c228e1
Face Landmark Estimation - http://www.csc.kth.se/~vahidk/papers/KazemiCVPR14.pdf
Face Landmark Python Code- https://gist.github.com/ageitgey/ae340db3e493530d5e1f9c15292e5c74
SVM- https://en.wikipedia.org/wiki/Support_vector_machine

Link to the Issue: ageitgey/face_recognition#333
The title of the issue better describes the issue the user faced. The user posted his work around clearly. Response to the issue is given in the form of a code. It is an effective solution proposed to the issue and the issue gets closed with a single response. Code snippet provided is very supportive for the issue to be resolved.
